### PR TITLE
docker-compose: storage overlays for minio/s3

### PR DESCRIPTION
**Pull request by mchalski**
Issues: MEN-915

@mendersoftware/rndity @maciejmrowiec @GregorioDiStefano

### CODE REVIEW

**Comment by bboozzoo** [comment]
It might make sense to add a helper script that includes proper compose files. I use this helper locally:

```
maciek@corsair:mender/integration (git)-[bboozzoo/minio-https] cat demo 
#!/bin/bash -x

exec docker-compose \
     -f docker-compose.yml \
     -f docker-compose.demo.yml \
     $*
```

Then `up` could call such script directly and so could the user.

**Comment by GregorioDiStefano** [comment]
@mchalski I have a PR open that has a similar change (independent S3 file), is it possible to just uncomment the key id and secret and you bash variables instead, as such: 

https://github.com/GregorioDiStefano/integration/blob/d90f14b4f4477d60bcd2a7d5b267f86154c0c5f6/docker-compose.s3.yml

this way I can just rebase my PR on yours, and use this.

**Comment by mchalski** [comment]
@GregorioDiStefano sure, done

File: docker-compose.demo.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ services:
         volumes:
             - ./ssl/storage-proxy/s3.docker.mender.io.crt:/var/www/storage-proxy/cert/cert.crt
             - ./ssl/storage-proxy/s3.docker.mender.io.key:/var/www/storage-proxy/cert/key.pem
+            - ./storage-proxy/nginx.conf:/usr/local/openresty/nginx/conf/nginx.conf
```
Comment (bboozzoo) [inline docker-compose.demo.yml]: nginx.conf is not part of demo setup, it's a part of minio backend compose

File: docker-compose.storage.minio.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+version: '2'
+services:
+
+    #
+    # minio
+    #
+    minio:
+        image: minio/minio:RELEASE.2016-12-13T17-19-42Z
+        networks:
+            mender:
```
Comment (bboozzoo) [inline docker-compose.storage.minio.yml]: just 

```
networks:
   - mender
``` 
aliases are demo only

File: docker-compose.storage.minio.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            #   512k - 512kB/s
+            DOWNLOAD_SPEED: 1m
+            MAX_CONNECTIONS: 100
+        depends_on:
+            - minio
+
+    #
+    # mender-deployments
+    #
+    mender-deployments:
```
Comment (bboozzoo) [inline docker-compose.storage.minio.yml]: this is demo only too

File: docker-compose.storage.minio.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    # rate & connection limiting
+    #
+    storage-proxy:
+        image: openresty/openresty:alpine
+        ports:
+            - "9000:9000"
+        networks:
+            mender:
+                aliases:
+                    - s3.docker.mender.io
```
Comment (bboozzoo) [inline docker-compose.storage.minio.yml]: this alias too, demo only the way i see it

File: docker-compose.demo.yml (hunk 1 above)
Comment (mchalski) [inline docker-compose.demo.yml]: duh, fixed

File: docker-compose.storage.minio.yml (hunk 3 above)
Comment (mchalski) [inline docker-compose.storage.minio.yml]: all aliases moved to demo layer, stacked a commit on top

**Comment by bboozzoo** [comment]
merge?

**Pull request merged**